OpenRA.Mods.Common/Activities/Move/Move.cs

```diff
@@ -248,7 +257,12 @@ public override bool Tick(Actor self)
 				// There is no point in waiting for the other actor to move if it is incapable of moving.
 				if (!mobile.CanEnterCell(nextCell, ignoreActor, BlockedByActor.Immovable))
 				{
-					path = EvalPath(BlockedByActor.Immovable);
+					path = EvalPath(BlockedByActor.Immovable).Path;
+
+					// If actor is blocked by immovable actors, suggest parent activiy to give up
+					if (path.Count == 0)
+						mobile.MoveResult = MoveResult.MovementStuck;
```

Comment:
We will probably need to teach `PopPath` to return something to indicate whether the move attempt should continue or be abandoned.

Currently PopPath returns null to indicate "There isn't a cell to move to right now, but try again in the future". We probably need a way to allowing the caller to know whether to try again in future or not.

Within this branch, where there in an immovable actor in the way *and* we couldn't find a path around, we want to signal there's no point trying again in the future, as we expect the block to be permenant. Therefore Tick can return true and set the MoveResult to `MoveResult.CompleteDestinationBlocked`.

The other branches in PopPath that return null because they want a temporary wait, e.g. for other blockers to move out the way, can allow Tick to return false and continue the Move.